spring-vault-core/src/test/java/org/springframework/vault/core/VaultSysTemplateIntegrationTests.java

```diff
@@ -98,6 +99,78 @@ public void mountShouldMountGenericSecret() {
 		assertThat(Arrays.asList("kv", "generic")).contains(secret.getType());
 	}
 
+	@Test
+	public void mountShouldMountKv1Secret() {
+
+		assumeTrue(vaultVersion.isGreaterThanOrEqualTo(
+			VaultRule.VERSIONING_INTRODUCED_WITH));
+
+		if (adminOperations.getMounts().containsKey("other/")) {
+			adminOperations.unmount("other");
+		}
+
+		VaultMount mount = VaultMount.builder().type("kv")
+			.config(Collections.singletonMap("default_lease_ttl",  "1h"))
+			.description("hello, world").build();
+
+		adminOperations.mount("other", mount);
+
+		Map<String, VaultMount> mounts = adminOperations.getMounts();
+
+		assertThat(mounts).containsKey("other/");
+
+		VaultMount kVv1 = mounts.get("other/");
+		assertThat(kVv1.getDescription()).isEqualTo(mount.getDescription());
+		assertThat(kVv1.getConfig()).containsEntry("default_lease_ttl", 3600);
+		assertThat(kVv1.getType()).isEqualTo("kv");
+
+		try {
+			// a versioned write (kv put) will fail for a kv (default version: 1, not versioned) store
+			vaultOperations.opsForVersionedKeyValue("other").put("secret",
+				Collections.singletonMap("key", "value"));
+
+			fail("Missing NullPointerException");
+		}
+		catch (NullPointerException e) {
+			assertThat(e).hasMessage(null);
```

Comment:
This is odd. Seeing a catch for `NullPointerException` rather requires a fix than an assertion.

Reply:
It is weird indeed, but I wasn't sure how to approach it.

The `put` call ends up returning no response body - it's a 204, and 3 headers are returned. As a result, an NPE comes from the `response` being `null` in `return getMetadata(response.getRequiredData());` in https://github.com/spring-projects/spring-vault/blob/master/spring-vault-core/src/main/java/org/springframework/vault/core/VaultVersionedKeyValueTemplate.java#L155.

There's nothing to build `Metadata` from. Would you like me to return an empty `Metadata` object in that case?

Reply:
So there are two things here: Using `opsForVersionedKeyValue` with `kv` version 1 is wrong per se because it's the wrong backend. In any case we should introduce a check and throw `IllegalStateException` if the response is null.  

For this test I suggest using the regular `VaultTemplate.write`/`read` methods. Seeing `opsForVersionedKeyValue`
with kv version 1 is misleading.

Reply:
@mp911de understood. My angle was such that one would expect kv-specific operations to work for a `kv` mount.
Will implemented as requested.